run_linux.go

```diff
@@ -66,6 +67,12 @@ func setChildProcess() error {
 
 // Run runs the specified command in the container's root filesystem.
 func (b *Builder) Run(command []string, options RunOptions) error {
+	// override contextdir from RunOptions if specified explicitly via --contextDir
+	// otherwise use ContextDir configured from build
+	if options.ContextDir != "" {
+		b.ContextDir = options.ContextDir
```

Comment:
This should _not_ quietly persist.  Go ahead and remove `Builder.ContextDir`.

Reply:
@nalind One problem with using it directly from `runOptions` would be that  i'll have to pass down ContextDir as an argument across all the function but since all the functions are already part of `builder` interface it would be easily accessible from builder property like `b.ContextDir`

Just thought this would be cleaner.

Reply:
The value should not be kept.